pkg/blockstorage/azure/auth.go

```diff
@@ -0,0 +1,121 @@
+package azure
+
+import (
+	"context"
+
+	"github.com/Azure/go-autorest/autorest/adal"
```

Comment:
Looks like this package will be [deprecated next year](https://pkg.go.dev/github.com/Azure/go-autorest/autorest/adal@v0.9.21#section-readme). Can we look into using [`azidentity`](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azidentity) instead?

Reply:
The purpose of interface `Authenticate()` is to validate Azure creds as a one-of action. These creds are then passed on to an `Authorizer()` that later authenticates and return a token for every Azure API calls that interact with Azure resources. 
And for `Authorizer()`, we are still using `adal` indirectly as part of  `github.com/Azure/go-autorest/autorest/azure/auth` 

Furthermore, `azidentity` and `adal` differs slightly in their credential definitions and workflows. So I'd prefer not to use `azidentity` for `Authenticate()` and `adal` for `Authorizer()` in case we don't completely understand the nuances in expectations.  

Instead, I had abstracted out the logics as much as I can, which should make APIs swapping easier during migration :)

Reply:
https://github.com/kanisterio/kanister/issues/1621